plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -343,6 +363,18 @@ func (cache *schedulerCache) cleanupAssumedPods(now time.Time) {
 		if !ok {
 			panic("Key found in assumed set but not in podStates. Potentially a logical error.")
 		}
+		var bindingDone bool
+		select {
+		case <-ps.waitsBinding:
+			bindingDone = true
+		default:
+			bindingDone = false
+		}
+		if !bindingDone {
+			glog.Warningf("Cant expire cache for pod %v/%v. Binding is still in progress.",
+				ps.pod.Namespace, ps.pod.Name)
+			continue
+		}
```

Comment:
OK - sorry. I'm doing few things at the same time and misread the condition.

Anyway - this doesn't solve the problem. If the cache will expire just after binding is finished (but not year read via watch), the same can happen. So introduce this logic doesn't make much sense, as the probability of it is not significantly lower, i think.